packages/shared_preferences/shared_preferences_android/android/src/main/kotlin/io/flutter/plugins/sharedpreferences/StringListObjectInputStream.kt

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.sharedpreferences
+
+import java.io.IOException
+import java.io.InputStream
+import java.io.ObjectInputStream
+import java.io.ObjectStreamClass
+
+/**
+ * An ObjectInputStream that only allows string lists, to prevent injected prefs from instantiating
+ * arbitrary objects.
+ */
+class StringListObjectInputStream(input: InputStream) : ObjectInputStream(input) {
+  @Throws(ClassNotFoundException::class, IOException::class)
+  override fun resolveClass(desc: ObjectStreamClass?): Class<*>? {
+    val allowList =
+        setOf(
+            "java.util.Arrays\$ArrayList",
+            "java.util.ArrayList",
+            "java.lang.String",
+            "[Ljava.lang.String;")
```

Comment:
Typo `Ljava`?

Reply:
Nope, `[Ljava.lang.String;` is the name of a the special class that specifically implements string arrays. This is an example of the potential fragility of this approach 😐

Reply:
OK after sleeping on it I think this approach makes sense. I am worried about the fragility but I think your error with desc.name contains the information we would need to parse if it came up. The better solutions from a dev perspective are still vulnerable so I think the allow list is what we have to go with.